src/core/data/transform.cc

```diff
@@ -301,6 +313,8 @@ void Transpose::print(std::ostream& out) const
 
 int32_t Transpose::target_ndim(int32_t source_ndim) const { return source_ndim; }
 
+void Transpose::return_promoted_dims(std::vector<int32_t>&) const {};
```

Comment:
I think we need to do something to handle this case, where dimensions are reordered. Consider that if dimension 1 of `A` is fake, then dimension 0 of `A.transpose([1,2,0])` is fake. See the mapper logs I get for this example:

```
out = cn.zeros((2,3,4))
x = cn.ones((4,3)) # shape=(4,3), promoted_dims=[]
y = x[:,cn.newaxis,:] # shape=(4,1,3), promoted_dims=[1]
z = y.transpose((1,2,0)) # shape=(1,3,4), promoted_dims=[0]
cn.exp(z, out=out)
```

```
MAP_TASK for cunumeric::UnaryOpTask(index_point=(0,1,2))<106> @ a.py:30
    Requirement[0](privilege=READ_ONLY,region=(2,(49,3),2),domain=<2,1>..<2,1>,fields=1000)
    Requirement[1](privilege=WRITE_DISCARD,region=(1,(16,1),1),domain=<0,1,2>..<0,1,2>,fields=1000)
MAP_TASK for cunumeric::UnaryOpTask(index_point=(1,1,2))<107> @ a.py:30
    Requirement[0](privilege=READ_ONLY,region=(2,(49,3),2),domain=<2,1>..<2,1>,fields=1000)
    Requirement[1](privilege=WRITE_DISCARD,region=(1,(17,1),1),domain=<1,1,2>..<1,1,2>,fields=1000)
```

Indeed, it is on dimension 0 of the launch space that the "collective" access of `x` is happening. 

I think you probably want to have processed the parent transform first, before you can shift the promoted dims around.

Reply:
I don't think there's anything to do for `Shift`. I am not 100% sure what needs to be done for `Delinearize`, I took a crack at it, we should get Wonchan's confirmation though.

Reply:
OK, I will ask Wonchan on slack.